test/unit/README.md

```diff
@@ -3,40 +3,105 @@
 
 ### Running unit tests
 
-1. via the command line
+1. Running just unit tests - `npm run test`
 
-```
-grunt test
-```
+2. Running coverage reporter - `npm run coverage`. 
+To check coverage, open `./coverage/lcov-report/index.html`.
 
-2. in the browser
-```
-open SpecRunner.html
-```
+3. Running tests in browser - `test-browser`.
 
 
-### Adding a new test file
+### Common concepts for writing tests
 
-New test files are added as a script tag to the SpecRunner.html
+1. Test suites should cover only public API.
 
-```html
-<script src="test/unit/application.app-regions.spec.js"></script>
-<script src="test/unit/application.spec.js"></script>
-```
+2. Code style.
 
+- Each `describe` should have name of tested method. 
+
+> There are should be not more then 1 nested describe for on method.
+ 
+Wrong way.
+ 
+ ```javascript
+ 
+ describe('some events in myMethod', function() {
+   describe('some logic', function() {
+     it('do something', function() {
+      ...  
+     })
+     
+     describe('some other logic', function() {
+       etc...
+     })
+   })
+ })
+```
 
-### Adding a new source file
 
-When you want to add a new source file to be tested, the file needs to be added in two places
+Wright way.
+
+
+```javascript
+ 
+  describe('#myMethod', function() {
+    describe('when some logic', function() {
+      it('should do something', function() {
+        ...  
+      })
+    })
+   
+    describe('when some other logic', function() {
+      it('should do something', function() {
+        ...  
+      })
+    })
+ })
+```
 
-1. SpecRunner.html
 
-```html
-<script src="src/item-view.js"></script>
+- `before/beforeEach` should consist only some preparation logic 
+but not call methods you expect to test.
+
+Wrong way.
+
+```javascript
+  describe('#myMethod', function() {
+    let myInstance;
+    
+    beforeEach(function() {
+      myInstance = new MyClass({
+        render: this.sinon.spy
+      });
+      myInstance.render();
+    })
+   
+    it('should do something', function() {
+      expect(myInstance.render).to.have.been.called;
+    })
+ })
 ```
 
-2. unit/setup/node.js file
+Wright way.
+
+```javascript
+  describe('#myMethod', function() {
+    let myInstance;
+    let renderSpy;
+    
+    beforeEach(function() {
+      renderSpy = this.sinon.spy();
+      
+      myInstance = new MyClass({
+        render: renderSpy
+      });
+    })
+   
+    it('should do something', function() {
+      myInstance.render();
+      
+      expect(renderSpy).to.have.been.called;
+    })
+ })
```

Comment:
sooo one other nit-pick style thing.. none of the closing `})`s have `;` and they probably should to match our specs.